site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/controller/SiteController.java

```diff
@@ -32,14 +32,20 @@ public SiteController(SiteService siteService) {
 
     @PreAuthorize("@authorisationService.authorise('create-site', #site.parentSiteId)")
     @PostMapping()
-    public ResponseEntity<Response> site(@RequestBody Site site) {
+    public ResponseEntity<Response> createSite(@RequestBody Site site) {
         String siteId = siteService.save(site);
         return ResponseEntity.status(HttpStatus.CREATED).body(new Response(siteId));
     }
 
+    @PreAuthorize("@authorisationService.authorise('view-site')")
     @PostAuthorize("@authorisationService.filterUserSites(returnObject.getBody(), #role)")
     @GetMapping
-    public ResponseEntity<List<Site>> sites(@RequestParam(value = "role", required = false) String role) {
+    public ResponseEntity<List<Site>> getSites(@RequestParam(value = "role", required = false) String role) {
+        return ResponseEntity.status(HttpStatus.OK).body(siteService.findSites());
+    }
+
+    @GetMapping("/authorize")
```

Comment:
It still need the PostAuthorize header that was here before:
maybe something like:
@PostAuthorize("@authorisationService.filterUserSites(returnObject.getBody(), null)")
Because it should only return sites user has a role assignment on

Reply:
ok..so when auth service calls the endpoint it could be from other endpoints that dont have the postAuthorize - if so, wouldnt it be better to add the filter method in authSevice rather than add an annotation which calls the auth service to the endpoint?

Reply:
So that is not what BAs wanted. Basically someone can just now call this endpoint and get all the sites, something I don't think we want to happen. Also I know remembered this endpoint is also used by the ui as part of story [ARTS-48](https://ndph-arts.atlassian.net/browse/ARTS-48) (by the way, this also means we need to change the path in the ui where it does the get my sites)